plugins/modules/helm_cli.py

```diff
@@ -0,0 +1,593 @@
+
+    rc, out, err = module.run_command(get_command)
+
+    if rc != 0:
+        module.fail_json(
+            msg="Failure when executing Helm command. Exited {0}.\nstdout: {1}\nstderr: {2}".format(rc, out, err),
+            command=get_command
+        )
+
+    # Helm 3 return "null" string when no values are set
+    if not is_helm_2 and out.rstrip("\n") == "null":
+        return {}
+    else:
+        return yaml.safe_load(out)
+
+
+# Get Release from all deployed releases
+def get_release(state, release_name, release_namespace):
+    if state is not None:
+        if is_helm_2:
+            for release in state['Releases']:
+                # release = {k.lower(): v for k, v in release.items()} # NOT WORKING wit python 2.6
+                release_lower = dict()
+                for k, v in release.items():
+                    release_lower[k.lower()] = v
+                release = release_lower
+                if release['name'] == release_name and release['namespace'] == release_namespace:
+                    return release
+        else:
+            for release in state:
+                if release['name'] == release_name and release['namespace'] == release_namespace:
+                    return release
+    return None
+
+
+# Get Release state from deployed release
+def get_release_status(command, release_name, release_namespace):
+    list_command = command + " list --output=yaml "
+
+    if not is_helm_2:
+        list_command += " --namespace=" + release_namespace
+        list_command += " --filter "
+
+    list_command += release_name
+
+    rc, out, err = module.run_command(list_command)
+
+    if rc != 0:
+        module.fail_json(
+            msg="Failure when executing Helm command. Exited {0}.\nstdout: {1}\nstderr: {2}".format(rc, out, err),
+            command=list_command
+        )
+
+    release = get_release(yaml.safe_load(out), release_name, release_namespace)
+
+    if release is None:  # not install
+        return None
+
+    release['values'] = get_values(command, release_name, release_namespace)
+
+    return release
+
+
+# Run Repo update
+def run_repo_update(command):
+    repo_update_command = command + " repo update"
+
+    rc, out, err = module.run_command(repo_update_command)
+    if rc != 0:
+        module.fail_json(
+            msg="Failure when executing Helm command. Exited {0}.\nstdout: {1}\nstderr: {2}".format(rc, out, err),
+            command=repo_update_command
+        )
+
+
+# Get chart info
+def fetch_chart_info(command, chart_ref):
+    if is_helm_2:
+        inspect_command = command + " inspect chart " + chart_ref
+    else:
+        inspect_command = command + " show chart " + chart_ref
+
+    rc, out, err = module.run_command(inspect_command)
+    if rc != 0:
+        module.fail_json(
+            msg="Failure when executing Helm command. Exited {0}.\nstdout: {1}\nstderr: {2}".format(rc, out, err),
+            command=inspect_command
```

Comment:
This will expose chart repo username and password to std output and/or log aggregators like splunk or logstash and can lead to CVE.

Reply:
I closed this conversation to keep only one with fabianvf